ibis/backends/sql/compilers/base.py

```diff
@@ -79,10 +82,18 @@ def __getattr__(self, name: str) -> Callable:
 
         __getitem__ = __getattr__
 
-    __slots__ = ("supports_filter",)
+    __slots__ = ("supports_filter", "supports_order_by", "requires_within_group")
 
-    def __init__(self, *, supports_filter: bool = False):
+    def __init__(
+        self,
+        *,
+        supports_filter: bool = False,
+        supports_order_by: bool = False,
+        requires_within_group: bool = False,
```

Comment:
I originally implemented things like this, but only snowflake would use this syntax for anything except group_concat (and that one had to be specialized in many backends anyway). Felt cleaner to split it out to a snowflake-specific subclass than add an extra kwarg here. Either works though 🤷.

Reply:
Hm, yeah, that might be better, let me fiddle with it a bit more. I think I've got everything passing except risingwave.